### PR TITLE
feat(workflow): API support for moving issues in and out of inbox

### PR DESCRIPTION
**Pull request by iProgramStuff**
Some basic API support for moving issues in and out of the inbox. 

Adds a "MANUAL" reason.

### CODE REVIEW

File: src/sentry/api/helpers/group_index.py [hunk 1]
```diff
@@ -976,4 +986,14 @@ def update_groups(request, projects, organization_id, search_fn):
             "children": [six.text_type(g.id) for g in groups_to_merge],
         }
 
+    # Support moving groups in or out of the inbox
+    inbox = result.get("inbox", None)
+    if inbox in (True, False):
```
Comment (wedamija) [inline src/sentry/api/helpers/group_index.py:991]: `inbox is not None` - It's already validated to be a boolean.

File: src/sentry/api/helpers/group_index.py [hunk 2]
```diff
@@ -976,4 +986,14 @@ def update_groups(request, projects, organization_id, search_fn):
...
+        if inbox is True:
```
Comment (wedamija) [inline src/sentry/api/helpers/group_index.py:992]: Nit: just `inbox`

File: src/sentry/api/helpers/group_index.py [hunk 3]
```diff
@@ -976,4 +986,14 @@ def update_groups(request, projects, organization_id, search_fn):
...
+            for group in group_list:
+                add_group_to_inbox(group, GroupInboxReason.MANUAL)
+        elif inbox is False:
```
Comment (wedamija) [inline src/sentry/api/helpers/group_index.py:995]: Nit: `not inbox`

**Review by wedamija — COMMENTED** [review]
lgtm, although not sure if you're intending to add things to `InboxDetailsValidator` in this pr?

**Comment by iProgramStuff** [comment]
> lgtm, although not sure if you're intending to add things to `InboxDetailsValidator` in this pr?

Probably not, not sure exactly how it will be used right now. It's probably only relevant for undo which is something we're only doing later

File: src/sentry/api/helpers/group_index.py [hunk 4]
```diff
@@ -976,4 +986,14 @@ def update_groups(request, projects, organization_id, search_fn):
...
         }
 
+    # Support moving groups in or out of the inbox
+    inbox = result.get("inbox", None)
+    if inbox is not None:
+        if inbox:
+            for group in group_list:
+                add_group_to_inbox(group, GroupInboxReason.MANUAL)
+        elif not inbox:
+            GroupInbox.objects.filter(group__in=group_ids).delete()
```
Comment (wedamija) [inline src/sentry/api/helpers/group_index.py:996]: Just checking you don't care about signals firing here.

**Review by wedamija — APPROVED** [review]

**Pull request merged**